--- pkg_rmleaves.orig	2014-02-19 13:58:20.969015298 -0600
+++ pkg_rmleaves	2014-02-19 13:57:29.478015196 -0600
@@ -61,17 +61,17 @@
 # update leaf files
 checkLeafs() {
 	if [ -n "$USE_PKGNG" ]; then
-		pkg query -e '%#r = 0' '"%n-%v" "%sh" "off" "%c"'
+		pkg query -e '%#r = 0' '#%n-%v# #%sh# #off# #%c#'
 	else
 		for i in "$PKGDIR/"*; do
 			if [ ! -s "$i/+REQUIRED_BY" ]; then 
 				pkg=$(basename "$i")
 				desc=''
 				[ -f "$i/+COMMENT" ] && desc=$(cat "$i/+COMMENT")
-				echo "\"$pkg\" \"\" \"off\" \"$desc\""
+				echo "#$pkg# ## #off# #$desc#"
 			fi
 		done
-	fi | sort > "$PKGFILE"
+	fi | sort | sed -e 's/"/\\"/g' | sed -e 's/#/"/g' > "$PKGFILE"
 
 	if [ -f "$PREV" ]; then
 		diff --unchanged-line-format='' --old-line-format='' --new-line-format='%L' "$PREV" "$PKGFILE" > "$TMPFILE"
